### PR TITLE
mgr/dashboard: NFS non-existent files cleanup

### PR DESCRIPTION
**Pull request by alfonsomthd**
After https://github.com/ceph/ceph/pull/42526 and https://github.com/ceph/ceph/pull/43725 merges,
the following files do not exist but there were still references to them:
- src/pybind/mgr/dashboard/services/ganesha.py
- qa/tasks/mgr/dashboard/test_ganesha.py

The following files were renamed but there were still references to old names:
- src/pybind/mgr/dashboard/controllers/nfsganesha.py:  nfsganesha.py --> nfs.py
- src/pybind/mgr/dashboard/tests/test_ganesha.py:  test_ganesha.py --> test_nfs.py

Other changes in qa/suites/rados/dashboard/tasks/dashboard.yaml:
- Add missing task: tasks.mgr.dashboard.test_api
- Sort dashboard tasks alphabetically.

Fixes: https://tracker.ceph.com/issues/53123
Signed-off-by: Alfonso Martínez <almartin@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [ ] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by avanthakkar — APPROVED** [review]

**Review by nizamial09 — APPROVED** [review]
LGTM

**Review by pereman2 — APPROVED** [review]

**Comment by alfonsomthd** [comment]
@jdurgin @neha-ojha Can you take a quick look at it?

**Comment by ljflores** [comment]
jenkins test dashboard

File: .github/labeler.yml [hunk 1]
```diff
@@ -246,10 +246,9 @@ nfs:
   - src/pybind/mgr/nfs/**
   - src/pybind/mgr/cephadm/services/nfs.py
   - src/pybind/mgr/cephadm/templates/services/nfs/ganesha.conf.j2
-  - src/pybind/mgr/dashboard/services/ganesha.py
-  - src/pybind/mgr/dashboard/tests/test_ganesha.py
+  - src/pybind/mgr/dashboard/controllers/nfs.py
+  - src/pybind/mgr/dashboard/tests/test_nfs.py
```
Comment (epuertat) [inline .github/labeler.yml:250]: ?
```suggestion
  - src/pybind/mgr/dashboard/tests/test_nfs.py
  - ./frontend/src/app/ceph/nfs/**
```

**Review by epuertat — APPROVED** [review]

**Pull request merged**